packages/client/src/clients/guide/client.ts

```diff
@@ -978,16 +991,22 @@ export class KnockGuideClient {
 
     this.knock.log(`[Guide] Handle Location change: ${href}`);
 
+    // If entering debug mode, fetch all guides.
+    const currentDebugParams = this.store.state.debug;
     const newDebugParams = detectDebugParams();
     this.setLocation(href, { debug: newDebugParams });
 
-    // If entering debug mode, fetch all guides.
-    const currentDebugParams = this.store.state.debug;
-    if (!currentDebugParams.forcedGuideKey && !!newDebugParams.forcedGuideKey) {
+    // If entering/exiting debug mode, refetch guides and resubscribe to the
+    // websocket channel.
+    if (
+      Boolean(currentDebugParams.forcedGuideKey) !==
+      Boolean(newDebugParams.forcedGuideKey)
+    ) {
       this.knock.log(
-        `[Guide] Entering debug mode for key: ${newDebugParams.forcedGuideKey}`,
+        `[Guide] ${newDebugParams.forcedGuideKey ? "Entering" : "Exiting"} debug mode`,
       );
       this.fetch();
+      this.resubscribe();
```

Comment:
```suggestion
      this.fetch();
      this.subscribe();
```
The subscribe() method already unsubscribes if there is an exiting `socketChannel`, so technically you could just call `this.subscribe()` here and it should do what `resubscribe()` is doing. If you want to make it extra clear, then you could call this.unsubscribe() and then this.subscribe(). All that is to say, not sure if we need the resubscribe() method. We don't provide `refetch()` either so I'd vote to hold off on adding that method.

Reply:
Ahh yeah, forgot about that check. I'll move to just use `subscribe()`